consensus/misc/eip4844/eip4844.go

```diff
@@ -81,6 +81,13 @@ func CalcExcessBlobGas(config *params.ChainConfig, parent *types.Header, headTim
 
 // CalcBlobFee calculates the blobfee from the header's excess blob gas field.
 func CalcBlobFee(config *params.ChainConfig, header *types.Header) *big.Int {
+	// OP-Stack chains don't support blobs, but still set the excessBlobGas field (always to zero).
+	// So this function is called in many places for OP-Stack chains too. In order to not require
+	// a blob schedule in the chain config, we short circuit here.
+	if config.IsOptimism() && config.BlobScheduleConfig == nil && header.ExcessBlobGas != nil && *header.ExcessBlobGas == 0 {
+		return minBlobGasPrice
+	}
```

Comment:
```suggestion
	if config.IsOptimism() {
		return minBlobGasPrice
	}
```
Why not simply like this?

The blob schedule shouldn't be set, but I imagine the blob schedule can be present accidentally even if blobs are not active at any fork because of the L1 difference.

 And the header value missing shouldn't mean that we continue to try and determine a blob fee.

Reply:
The reason to be stricter here from my side was to catch any _unexpected_ values. An OP-Stack chain with Cancun active should guarantee non-nil `header.ExcessBlobGas` field with value 0 and a nil blob schedule. If any of these assumptions fail, I'd rather have the program panic, because it would mean an error in the setup of the program. I feel like it's better to test all assumptions rather than silently short-circuit with a weird setup.

But I don't feel strongly. If you prefer the simple catch-call check, I'm happy with it too.

Reply:
We discussed in DM to add an explicit panic check